lib/stripe/plans.rb

```diff
@@ -80,20 +79,14 @@ def product_options
         product_id.presence || { name: name, statement_descriptor: statement_descriptor }
       end
 
+      def create_options_with_products
+        default_create_options.merge({ product: product_options })
+                              .delete_if{|_, v| v.nil? }
+      end
+
       def create_options_without_products
-        {
-          currency: currency,
-          name: name,
-          amount: amount,
-          interval: interval,
-          interval_count: interval_count,
-          trial_period_days: trial_period_days,
-          metadata: metadata,
-          usage_type: usage_type,
-          aggregate_usage: aggregate_usage,
-          billing_scheme: billing_scheme,
-          statement_descriptor: statement_descriptor
-        }.delete_if{|_, v| v.nil? }
+        default_create_options.merge(product_options)
```

Comment:
though unlikely, I'm a bit worried that `product_id` (see code snippet below) might leak into this options hash so I'm going to revert this commit.

https://github.com/tansengming/stripe-rails/blob/6bf308b51dd65e6fd08084a1e21fdcd40701e1f5/lib/stripe/plans.rb#L76-L78